src/compiler/checker.ts

```diff
@@ -33939,6 +33940,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
     }
 
+    function getEffectiveCheckArgumentNode(argument: Expression): Expression {

```

Comment:
I was looking for other solutions that wouldn't explicitly target argument nodes. However, I concluded that this is quite a specific situation because it's the only situation (that I could find) when the `SatisfiesExpression` node is **also** the location on which other errors (argument checks) are meant to be reported. Usually the "final target" (like an assignment target) of `SatisfiesExpression` is different so there is no such overlap~ like here.

Reply:
> Usually the "final target" (like an assignment target) of SatisfiesExpression is different so there is no such overlap~ like here.

Counterpoint:
```ts
((): null => ({}) satisfies unknown)();
```

Reply:
Food for thought: If the arrow function above is rewritten with braces, then the error span is correctly placed on the `return`.  For braceless arrow functions the "final target" could be the `=>` token.

Reply:
> Counterpoint:

Great example :P I'll fix this in a moment

> For braceless arrow functions the "final target" could be the => token.

I prefer to highlight smaller spans over huge ones but this one feels just too short. OTOH, an elementwise elaboration can highlight even a single character if a property is that short. I'm not sure if it would be completely apparent to everybody that highlighted `=>` is about the return value. It's not that it doesn't make sense when explained - it's just that in isolation it doesn't feel completely intuitive/unambiguous